app/src/main/java/eu/kanade/tachiyomi/ui/more/stats/StatsPresenter.kt

```diff
@@ -0,0 +1,86 @@
+import eu.kanade.tachiyomi.data.database.models.LibraryManga
+import eu.kanade.tachiyomi.data.database.models.Manga
+import eu.kanade.tachiyomi.data.database.models.Track
+import eu.kanade.tachiyomi.data.download.DownloadManager
+import eu.kanade.tachiyomi.data.preference.MANGA_HAS_UNREAD
+import eu.kanade.tachiyomi.data.preference.MANGA_NON_COMPLETED
+import eu.kanade.tachiyomi.data.preference.MANGA_NON_READ
+import eu.kanade.tachiyomi.data.preference.PreferencesHelper
+import eu.kanade.tachiyomi.data.track.TrackManager
+import eu.kanade.tachiyomi.data.track.TrackService
+import eu.kanade.tachiyomi.source.CatalogueSource
+import eu.kanade.tachiyomi.source.SourceManager
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.ui.more.stats.StatsHelper.getReadDuration
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+/**
+ * Presenter of [StatsController].
+ */
+class StatsPresenter(
+    private val db: DatabaseHelper = Injekt.get(),
+    private val prefs: PreferencesHelper = Injekt.get(),
+    private val trackManager: TrackManager = Injekt.get(),
+    private val downloadManager: DownloadManager = Injekt.get(),
+    private val sourceManager: SourceManager = Injekt.get(),
+) {
+
+    private val libraryMangas = getLibrary()
+    val mangaDistinct = libraryMangas.distinct()
+
+    private fun getLibrary(): MutableList<LibraryManga> {
+        return db.getLibraryMangas().executeAsBlocking()
+    }
+
+    fun getTracks(manga: Manga): MutableList<Track> {
+        return db.getTracks(manga).executeAsBlocking()
+    }
+
+    fun getLoggedTrackers(): List<TrackService> {
+        return trackManager.services.filter { it.isLogged }
+    }
+
+    fun getSources(): List<CatalogueSource> {
+        val languages = prefs.enabledLanguages().get()
+        val hiddenCatalogues = prefs.hiddenSources().get()
+        return sourceManager.getCatalogueSources()
+            .filter { it.lang in languages }
+            .filterNot { it.id.toString() in hiddenCatalogues }
+    }
+
+    fun getGlobalUpdateManga(): Map<Long?, List<LibraryManga>> {
+        val includedCategories = prefs.libraryUpdateCategories().get().map(String::toInt)
+        val excludedCategories = prefs.libraryUpdateCategoriesExclude().get().map(String::toInt)
+        val restrictions = prefs.libraryUpdateMangaRestriction().get()
+        return libraryMangas.groupBy { it.id }
+            .filterNot { it.value.any { manga -> manga.category in excludedCategories } }
+            .filter { includedCategories.isEmpty() || it.value.any { manga -> manga.category in includedCategories } }
+            .filterNot {
+                val manga = it.value.first()
+                MANGA_NON_COMPLETED in restrictions && manga.status == SManga.COMPLETED ||
+                    MANGA_HAS_UNREAD in restrictions && manga.unread != 0 ||
+                    MANGA_NON_READ in restrictions && manga.totalChapters > 0 && !manga.hasRead
```

Comment:
I'd wrap each && logic in these with parenthesis, readability is clear but i dont trust kotlin sometimes without them